src/borg/remote.py

```diff
@@ -1194,21 +1206,30 @@ def cache_if_remote(repository, *, decrypted_cache=False, pack=None, unpack=None
         raise ValueError('decrypted_cache and pack/unpack/transform are incompatible')
     elif decrypted_cache:
         key = decrypted_cache
-        cache_struct = struct.Struct('=I')
+        # 32 bit csize, 64 bit (8 byte) xxh64
+        cache_struct = struct.Struct('=I8s')
         compressor = LZ4()
 
         def pack(data):
-            return cache_struct.pack(data[0]) + compressor.compress(data[1])
+            csize, decrypted = data
+            compressed = compressor.compress(decrypted)
+            return cache_struct.pack(csize, xxh64(compressed)) + compressed
 
         def unpack(data):
-            return cache_struct.unpack(data[:cache_struct.size])[0], compressor.decompress(data[cache_struct.size:])
+            data = memoryview(data)
+            csize, checksum = cache_struct.unpack(data[:cache_struct.size])
+            compressed = data[cache_struct.size:]
+            if checksum != xxh64(compressed):
+                logger.warning('Repository metadata cache: detected corrupted data in cache!')
+                raise RepositoryCache.InvalidateCacheEntry
```

Comment:
can we name the exception rather after the exceptional condition encountered (`InvalidCacheEntry`) rather than what we intend to do in the exception handler (invalidate/remove that entry).